src/oop/special_class_method/medium/q7/MainTest.java

```diff
@@ -0,0 +1,205 @@
+        };
+        Class<?>[] methodModifierClasses = {
+                int.class
+        };
+        Object output = testClass.callMethod(setAttributeMethodName1, setMethodArguments, testInstance);
+        assertTrue(testClass.hasModifier(setAttributeMethodName1, methodModifierClasses, "public"),
+                "Your " + setAttributeMethodName1 + " method does not have the correct visibility modifier.");
+        String incorrectSetterMessage =
+                "Your " + setAttributeMethodName1 + " method does not correctly update the value of " + attributeName1 + ".";
+        _assertEquals(updatedValue, testClass.getFieldValue(testInstance, attributeName1),
+                incorrectSetterMessage);
+        assertNull(output, String.join(" ", "Your", setAttributeMethodName1, "method should not return any output"));
+    }
+
+    @ParameterizedTest
+    @MethodSource("intInputProvider")
+    public void correctSetNumNickelsMethod(int updatedValue) throws Throwable {
+        Object[][] instantiationArguments = {
+        };
+        Object testInstance = testClass.createInstance(instantiationArguments);
+        Object[][] setMethodArguments = {
+                {updatedValue, int.class}
+        };
+        Class<?>[] methodModifierClasses = {
+                int.class
+        };
+        Object output = testClass.callMethod(setAttributeMethodName2, setMethodArguments, testInstance);
+        assertTrue(testClass.hasModifier(setAttributeMethodName2, methodModifierClasses, "public"),
+                "Your " + setAttributeMethodName2 + " method does not have the correct visibility modifier.");
+        String incorrectSetterMessage =
+                "Your " + setAttributeMethodName2 + " method does not correctly update the value of " + attributeName2 + ".";
+        _assertEquals(updatedValue, testClass.getFieldValue(testInstance, attributeName2),
+                incorrectSetterMessage);
+        assertNull(output, String.join(" ", "Your", setAttributeMethodName1, "method should not return any output"));
+    }
+
+    private static Stream<Arguments> intInputProvider2() {
+        return Stream.of(
+                Arguments.of(2, 3),
+                Arguments.of(32, 1323),
+                Arguments.of(43, 213),
+                Arguments.of(124332, 434343),
+                Arguments.of(223, 0)
+
+        );
+    }
+
+
+    @ParameterizedTest
+    @MethodSource("intInputProvider2")
+    public void correctToStringMethod(int value1, int value2) throws Throwable {
+        Object[][] arguments1 = {
+                {value1, int.class}
+        };
+        Object[][] arguments2 = {
+                {value2, int.class}
+        };
+        double value = (value1 * 1 + value2 * 5) / 100.0;
+        String incorrectToStringMessage = "Your toString method does not return the correct String.";
+        Object testInstance = testClass.createInstance(null);
+        testClass.callMethod(setAttributeMethodName1, arguments1, testInstance);
```

Comment:
switch the call methods to setField that way if the set method is incorrect it doesn't interrupt this test